### PR TITLE
Rda refacto state initialisation

### PR DESCRIPTION
**Pull request by Pamplemousse**
Make it easier to generate initialisation states, in particular, initialisation for a `SliceToSink` is coming.

Needs: #1873 !

### CODE REVIEW

File: angr/analyses/reaching_definitions/live_definitions.py [hunk 1]
```diff
@@ -65,15 +60,30 @@ def __init__(self, arch, track_tmps=False, analysis=None, init_func=False, cc=No
...
+                len(self.stack_definitions),
+                len(self.memory_definitions),
+                )
         if self._track_tmps:
             ctnt += ", %d tmpdefs" % len(self.tmp_definitions)
         return "<%s>" % ctnt
 
-    def _init_func(self, cc, func_addr):
+    def _set_initialisation_values(self, subject, rtoc_value=None):
+        if subject.type == SubjectType.Function:
```
Comment (ltfish) [inline angr/analyses/reaching_definitions/live_definitions.py]: Use `is` to compare types.

File: angr/analyses/reaching_definitions/live_definitions.py [hunk 2]
```diff
@@ -65,15 +60,30 @@ def __init__(self, arch, track_tmps=False, analysis=None, init_func=False, cc=No
...
+            if isinstance(self.arch, archinfo.arch_ppc64.ArchPPC64) and not rtoc_value:
+                raise ValueError('The architecture being ppc64, the parameter `rtoc_value` should be provided.')
+
+            self._initialise_function(
+                subject.cc,
+                subject.content.addr,
+                rtoc_value,
+            )
+        elif subject.type == SubjectType.Block:
```
Comment (ltfish) [inline angr/analyses/reaching_definitions/live_definitions.py]: Same here.

File: angr/analyses/reaching_definitions/def_use_graph.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import networkx
+
+from .definition import Definition
+
+
+def _is_definition(node):
+    return isinstance(node, Definition)
+
+
+class DefUseGraph():
```
Comment (ltfish) [inline angr/analyses/reaching_definitions/def_use_graph.py]: Parentheses are redundant here.

Comment (Pamplemousse) [reply angr/analyses/reaching_definitions/def_use_graph.py]: This is on top of #1869 , will be corrected when rebasing.

**Comment by ltfish** [comment]
Can you please rebase it?

**Comment by ltfish** [comment]
Tests fail!

**Comment by Pamplemousse** [comment]
Oh yes, that's expected now:
Because I force the initialisation of certain register with a code location of `ExternalCodeLocation`, the comparison to the pickle data is failing for this attribute.

However, I regenerating the pickled dump blindly is a good idea.
I am taking a bit of time to make sure that only this attribute differs.
(which is not straightforward given the low quality implementation of this test...)

**Comment by ltfish** [comment]
Thanks!

**Pull request merged**